modules/examples/sources/random-choice-generator/random-choice-generator.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2023 Attila Szakacs
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include <string>
+#include <unistd.h>
+#include <vector>
+
+#include "random-choice-generator.hpp"
+
+#define get_RandomChoiceGeneratorCpp(s) (((RandomChoiceGeneratorSourceDriver *) s)->cpp)
+
+typedef struct RandomChoiceGeneratorSourceDriver_ RandomChoiceGeneratorSourceDriver;
+
+class RandomChoiceGeneratorCpp
+{
```

Comment:
Maybe we could derive this from the source driver directly.

Maybe we would have an issue with the destructor side (as we need to be able to free this from the c side as the ref count gets to zero).

Reply:
I came to the conclusion that there would be problems on both ctor and dtor side if we derived directly, but it was a busy day, so I will think this through again tomorrow.

Reply:
g++ and gcc pack structs differently.
I haven't checked the standard yet, but inheriting a C struct from C++ is not straight-forward as far as I remember.